src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -124,12 +139,17 @@ private static Long getTheme(String widget) {
         }
 
         // mostly copied from the javadoc for ReentrantReadWriteLock
-        Long theme = widgetToTheme.get(widget);
+        Long theme = null;
+
+        if (dpiAwareWidgetToTheme.get(dpi) != null)
+        {
+            theme = dpiAwareWidgetToTheme.get(dpi).get(widget);
```

Comment:
avoid calling `dpiAwareWidgetToTheme.get(dpi)` twice. Assign to a variable and reuse